### ISSUE TITLE
Refactor `fromBaseUnit` usage

### ISSUE THREAD

**Issue by 0xApotheosis** [shapeshift/web#1148]
Would it make sense to accept `BigNumber.Value | null | undefined` as a value in `fromBaseUnit`? 
https://github.com/shapeshift/web/blob/433be3392ff233bdaf1141f73d44fe67f68aef09/src/lib/math.ts#L6

Since we're using `bnOrZero`, IMO we should just use the same input types that it accepts:
https://github.com/shapeshift/web/blob/433be3392ff233bdaf1141f73d44fe67f68aef09/src/lib/bignumber/bignumber.ts#L8

This would allow us to avoid using these kind of fallbacks

_Originally posted by @gomesalexandre in https://github.com/shapeshift/web/pull/1124#discussion_r821197379_

**Comment by mrnerdhair** [comment]
IMO that's the wrong fallback. It should be more like 

```
<Amount.Crypto
  value={txDetails.sellTx?.value ? fromBaseUnit(txDetails.sellTx?.value, txDetails.sellAsset?.precision) : '0'}
  symbol={txDetails.sellAsset?.symbol ?? ''}
  maximumFractionDigits={6}
/>
```

(And that's still questionable -- we shouldn't really ever want to render that component if the info it's supposed to be showing isn't present.)

The type for `TxDetails` should be updated to exclude error states like `txDetails.sellTx.value` being present but `txDetails.sellAsset.precision` somehow being missing. The fields in that type are only optional because `useTxDetails()` is ignoring proper error handling and making it the client's job. (Contrary to apparently-popular opinion, optional chaining does not magically solve everything.)

TL;DR:  Input types should always be as normalized as possible, and shouldn't be widened simply because the current implementation of a function happens to be able to handle the wider type. The more we indulge that sort of thing, the more edge cases we'll have to deal with in inappropriate places and the harder it will ever be to refactor things to do it right way.

**Comment by 0xean** [comment]
@0xApotheosis / @gomesalexandre  to sync with @mrnerdhair on the correct path forward.

**Comment by gomesalexandre** [comment]
Closing as per agreement with @mrnerdhair, following the discussion in this issue on normalization.
